full-service/src/db/account.rs

```diff
@@ -574,6 +578,26 @@ impl AccountModel for Account {
 
         Ok(highest_subaddress_index as u64 + 1)
     }
+
+    fn view_private_key(self) -> Result<RistrettoPrivate, WalletDbError> {
+        if self.view_only {
+            let view_account_key: ViewAccountKey = mc_util_serial::decode(&self.account_key)?;
+            return Ok(*view_account_key.view_private_key());
+        }
+
+        let account_key: AccountKey = mc_util_serial::decode(&self.account_key)?;
+        return Ok(*account_key.view_private_key());
```

Comment:
Can this be simplified and instead call `self.view_account_key()?.view_private_key()`?